Add offset value to `ParseToken`

shiro opened the pull request:
## Description

Adds an `offset` value to parsed tokens for easier post-processing by being able to find the strings in the source text.

I plan to integrate this into https://github.com/shikijs/shiki/blob/f8ef446af3f5f3910131b692e0c81ce032754ffa/packages/core/src/highlight/code-to-tokens-ansi.ts#L62 once it gets merged.

## Changes

- adds `offset` to `ParseToken`
- updates readme
- adds some comments that the TS LSP can show on "quick definition" popups in most IDEs

blake-mealey commented on src/parser.ts:
```diff
@@ -172,6 +178,7 @@ export function createAnsiSequenceParser() {
             foreground,
             background,
             decorations: new Set(decorations),
+            offset: position,
```
This will be the position of the current parse call, but not the beginning of the full input if it is being parsed in chunks. I think the offset should be maintained at the parser level

blake-mealey reviewed (commented):
Thanks for the PR! Added a comment

blake-mealey commented:
Also please add some tests

shiro commented:
> I think the offset should be maintained at the parser level

Agreed, updated the code.

> Also please add some tests

Added a chunk test and updated other test snapshots, let me know if it's sufficient.
